src/kibana/plugins/discover/components/field_chooser/field_chooser.html

```diff
@@ -21,8 +21,11 @@
       <button
         ng-class="{ 'btn-default': !filter.active, 'btn-success': filter.active, 'hidden-xs': !showFields, 'hidden-sm': !showFields }"
         class="btn btn-xs btn-default pull-right discover-field-filter-toggle"
-        ng-click="showFilter = !showFilter">
-        <i class="fa fa-gear"></i>
+        ng-click="showFilter = !showFilter"
+        aria-label="Field Settings"
+        aria-haspopup="true"
+        aria-expanded="{{ showFilter ? 'true' : 'false' }}">
+        <i aria-hidden="true" class="fa fa-gear"></i>
```

Comment:
You need to have the `aria-hidden` if the label is on the outer element as I understand it. If the label is on the inner element you wouldn't need hidden as it would override whatever the screenreader would try to say for that item. With that said it makes more sense to hide the icon and label the button which would be clicked so they attempt to interact with the correct element, not just a random icon.

`aria-hidden` though is necessary if the item (the `<i>` in this case) doesn't have `aria-label` on it.